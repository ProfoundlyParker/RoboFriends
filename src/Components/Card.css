.background {
    background-color: #78cdfe;
}