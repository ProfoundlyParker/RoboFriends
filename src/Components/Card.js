import React from 'react';

const Card = ({ firstName, lastName, email, image }) => {
    return (
        <div className='tc bg-light-green dib br3 ma2 grow bw2 shadow-5'>
            <img src={image} alt="robots"></img>
         <div>
            <h2>{firstName} {lastName}</h2>
            <p>{email}</p>
         </div>
        </div>
    );
}

export default Card;
