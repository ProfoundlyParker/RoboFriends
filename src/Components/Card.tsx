import React from 'react';

// Added types for card props received
type CardProps = {
    firstName: string;
    lastName: string;
    email: string;
    image?: string;
}

// Displays robot cards with personal info and images
const Card: React.FC<CardProps> = ({ firstName, lastName, email }) => {
    // Displays robot images instead of cat images
    const robotImage = `https://robohash.org/${email}?set=set1`
    return (
        <div className='text-center bg-card-bg inline-block rounded-2xl m-2 grow border-0 shadow-2xl'>
            <img src={robotImage} height='300px' width='300px' alt="robots"></img>
         <div>
            <h2 className='block box-border text-2xl font-bold mt-5 ms-0 me-0 m-4'>{firstName} {lastName}</h2>
            <p className='box-border block ms-0 me-0 m-4 mt-5 mb-4'>{email}</p>
         </div>
        </div>
    );
}

export default Card;
