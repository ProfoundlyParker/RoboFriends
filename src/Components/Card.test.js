import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

it("expect to snapshot Card component", () => {
  const { asFragment } = render(<Card />);
  expect(asFragment()).toMatchSnapshot();
});
