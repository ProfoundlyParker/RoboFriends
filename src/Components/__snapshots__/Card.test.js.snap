// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to snapshot Card component 1`] = `
<DocumentFragment>
  <div
    class="text-center bg-card-bg inline-block rounded-2xl m-2 grow border-0 shadow-2xl"
  >
    <img
      alt="robots"
      height="300px"
      width="300px"
    />
    <div>
      <h2
        class="block box-border text-2xl font-bold mt-5 ms-0 me-0 m-4"
      >
         
      </h2>
      <p
        class="box-border block ms-0 me-0 m-4 mt-5 mb-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
